app/main.cpp

```diff
@@ -230,6 +230,11 @@ int main(int argc, char *argv[])
                                 QLatin1String("max-stack"), QLatin1String("127"));
     parser.addOption(maxStack);
 
+    QCommandLineOption branchTraverse(QLatin1String("branch-traverse"),
+                                      QCoreApplication::translate(
+                                              "main", "Short branchStack resolveCallchain traverse. Concerns lbr."));
```

Comment:
This description is not meaningful, what does this mean? can you please rephrase?

Also, please don't use the capitalization from the code in user-facing strings, i.e. "branch stack", "resolve callchain", "LBR".

Reply:
I also don't get why one would _not_ want to traverse the LBR call chain? again, this needs a better explanation in the commit message for why you would want this.

to me it sounds totally bogus - if I record with LBR I want to traverse the LBR chain. Otherwise I don't get anything useful at all, so I could just as well record without LBR in the first place?

Reply:
For the moment (without the changes) in resolveCallchain there is the 'break' from callchain traverse if it hasBranchStack. 
And callchain for LBR in this case is looked only for branchStack. 

It is not enough for me and LBR Disassembly events costs. So I traverse additionally some frames from LBR callchain.
The option 'branch-traverse' just turns off these additional traverses and traverse LBR calchain as is now from branchStack..

Reply:
as I said multiple times - I do not follow you at all on this. what have the LBR contents to do with disassembly?

We just need a callchain, which we can get through either fp/dwarf/LBR - it doesn't matter what is used behind the scenes.

Then for every symbol we need to know the start and name such that we can pass it to objdump later on, right?

The IP for per-instruction cost annotation is stored separately in `PerfRecordSample::ip()`

Are you maybe trying to get loop-count costs or something like that via LBR here? If so, why should that be optional - we would always want that. But that would require a separate kind of entry in our callchain then, or how are you handling this?

Please - again - be more specific! Give examples!

Reply:
Ok. I don't insist the option. I have entered it just to control impact to performance. I will move it out of merge request.